pkg/agent/cniserver/server.go

```diff
@@ -461,6 +463,15 @@ func (s *CNIServer) CmdAdd(ctx context.Context, request *cnipb.CniCmdRequest) (*
 	klog.Infof("CmdAdd for container %v succeeded", cniConfig.ContainerId)
 	// mark success as true to avoid rollback
 	success = true
+
+	if features.DefaultFeatureGate.Enabled(features.SecondaryInterfaceSupport) {
```

Comment:
Could we add and pass a boolean like secInterfaceEnabled to CNIServer, instead of checking the feature gate inside the CNIServer code? I would avoid featuregate in all code.

Reply:
Or based on whether the cniPodInfoStore argument passed to Initialize() is nil, we can save the value to a boolean flag of CNIServer struct, like cacheCNIPodInfo, and check s.cacheCNIPodInfo here.